src/main/java/io/jenkins/plugins/jwt_auth/JwtAuthSecurityRealm.java

```diff
@@ -266,6 +275,25 @@ private Authentication getAuthFromToken(ServletRequest servletRequest) {
 					// put it in our "cache"
 					userToGroupsCache.put(username, grantedGroups);
 
+
+
+                    User user = User.getById(username, true);
+
+					if(fullNameClaim != null) {
+						String fullName = jwtClaims.getClaimValueAsString(fullNameClaim);
+						if (fullName != null) {
+							user.setFullName(fullName);
+						}
+					}
+
+					if(emailClaimName != null) {
+						String email = jwtClaims.getClaimValueAsString(emailClaimName);
+						if (email != null) {
+							user.addProperty(new Mailer.UserProperty(email));
+						}
+					}
```

Comment:
The logic in this block of code will never _remove_ the email & fullname from a user if the JWT user details removed them, which would then lead to inconsistency (as users who'd logged in when JWT provided the data would keep the emails/names that JWT used to provide, but users who were newly logged-in wouldn't have them).
That kind of "you're seeing old data" issue would make it hard to debug as it might be days/months before a lack of data was discovered.

MIO it would be better if these fields are _always_ set (or unset) when the jwt-auth-plugin has been configured to set them.

i.e. if fullNameClaim != null (and isn't empty) then user.setFullName should always be called, either with the data that the jwtClaims object contained, or with null/empty-string if JWT didn't provide data.
Ditto emailClaimName.

That'd mean that, if an admin configures this plugin to grab the name & email from the JWT data then it'll _always_ do that, and any problems with doing that will be immediately visible rather than leaving "what Jenkins knew from before" in place.

Reply:
@Peter-Darton-i2 I don't entirely disagree with you, however @narcoticfresh previous comment is a bit contrary to this. I could go either way, however for now I'll implement per maintainer request. If it needs further dialog, I'm open to discussion.
> make sure you only do User.getById(username, true); and user.save() when it's actually necessary. There is no need to get and save it when both new settings are unconfigured.